src/packages/DAO/elections.dao.ts

```diff
@@ -38,31 +44,55 @@ export async function createElection(formData: CreateElection) {
 export async function startElection(formData: FormData) {
   const electionId = String(formData.get("electionId"));
   const updateElectionSchema = z.string().uuid();
-  const parsedElectionId = updateElectionSchema.parse(electionId);
+  const parsedElectionId = updateElectionSchema.safeParse(electionId);
+
+  if (!parsedElectionId.success) {
+    throw new BadRequestError(parsedElectionId.error.name);
+  }
 
   await prisma.election.update({
-    where: { id: parsedElectionId },
+    where: { id: parsedElectionId.data },
     data: {
       name: "Updated on server",
       status: ElectionStatus.ONGOING,
     },
   });
 
-  revalidatePath("/");
+  redirect(`http://localhost:3000/elections/${electionId}`);
+}
+
+async function getElections() {
+  const res = await prisma.election.findMany();
+  if (!res) {
+    throw new NotFound("");
+  }
+  return res;
+}
+
+async function findByNameOrStatus({ name, status }: FindElection) {
+  const resultElection = await getElections();
+  const result = resultElection.find(
+    (e) => e.name === name && e.status === status,
+  );
+  return result;
 }
 
 export async function addElection(formData: FormData) {
   const data = Object.fromEntries(formData);
-  const parsedData = createElectionValidator.parse(data);
+  const parsedData = createElectionValidator.safeParse(data); // Error empty input
+  if (!parsedData.success) {
+    throw new BadRequestError(parsedData.error.name);
+  }
+  const electionExist = await findByNameOrStatus(parsedData.data); // Error duplicate data
```

Comment:
As a general rule, if we have a `await` keyword, we also need a `try/catch`, as  long-running tasks can fail for various reasons.

```suggestion
  try {
    const electionExist = await findByNameOrStatus(parsedData.data); // Error duplicate data
  catch (err) {
    // handle error here ...
  }
```

or you can do it directly on the `Promise`:
```suggestion
  const electionExist = await findByNameOrStatus(parsedData.data).catch((err: Error) => {
    console.warn(err)
    return nil
  })
```

Reply:
The code has been refactored. Thank you :)